pkg/uninstaller/resources.go

```diff
@@ -0,0 +1,198 @@
+	r1 := client.Resource(clusterconfigV1alpha1.ClusterConfigGroupVersionResource)
+	t, err := r1.Namespace("").List(context.TODO(), metav1.ListOptions{TypeMeta: metav1.TypeMeta{}})
+	if err != nil {
+		return err
+	}
+	klog.V(8).Infof("Getting clusterConfigs")
+	var clusterconfigs clusterconfigV1alpha1.ClusterConfigList
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(t.UnstructuredContent(), &clusterconfigs); err != nil {
+		return err
+	}
+	klog.V(8).Infof("Patching ClusterConfigs")
+	for _, item := range clusterconfigs.Items {
+		patch := []byte(`{"spec": {"advertisementConfig": { "outgoingConfig" : { "enableBroadcaster" : false}}}}`)
+		_, err = r1.Namespace(item.Namespace).Patch(context.TODO(), item.Name, types.MergePatchType, patch, metav1.PatchOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func CheckReconciliation(client dynamic.Interface, toCheck toCheckDeleted) (bool, error) {
+	var (
+		objectList  *unstructured.UnstructuredList
+		err         error
+		labelString string
+	)
+
+	if labelString, err = generateLabelString(toCheck.labelSelector); err != nil {
+		return false, err
+	}
+	options := metav1.ListOptions{
+		LabelSelector: labelString,
+	}
+
+	objectList, err = client.Resource(toCheck.gvr).Namespace("").List(context.TODO(), options)
+
+	if apierrors.IsNotFound(err) {
+		return true, nil
+	}
+
+	if err != nil {
+		return false, err
+	}
+
+	if len(objectList.Items) == 0 {
+		klog.V(6).Infof("%s not found", toCheck.gvr)
+		return true, nil
+	}
+
+	return false, nil
+
+}
+
+func WaitForResources(client dynamic.Interface) error {
+	klog.Info("Waiting for Liqo Resources to be correctly deleted")
+	var wg sync.WaitGroup
+	result := make(chan *resultType, len(toCheck))
+	wg.Add(len(toCheck))
+	for _, resource := range toCheck {
+		go WaitForCompletion(client, resource, result, &wg)
+	}
+	for elem := range result {
+		if !elem.Success {
+			klog.Errorf("Error while waiting for %s to be deleted", elem.Resource.gvr.GroupResource())
+			return nil
+		}
+		printLabels, _ := generateLabelString(elem.Resource.labelSelector)
+		klog.Infof("%s instances with \"%s\" labels correctly deleted", elem.Resource.gvr.GroupResource(), printLabels)
+	}
+	wg.Wait()
+	close(result)
+	return nil
+}
+
+func WaitForCompletion(client dynamic.Interface, toCheck toCheckDeleted, result chan *resultType, wg *sync.WaitGroup) {
+	defer wg.Done()
+	ticker := time.NewTicker(5 * time.Second)
+	quit := make(chan struct{})
+	var wError error
+	value := false
+	var res = &resultType{
+		Resource: toCheck,
+		Success:  false,
+	}
+	timeout := time.NewTicker(1 * time.Minute)
+	for {
+		select {
+		case <-timeout.C:
+			close(quit)
+		case <-ticker.C:
+			value, wError = CheckReconciliation(client, toCheck)
+			if value {
+				res.Success = true
+				close(quit)
+			} else if wError != nil {
+				close(quit)
+			}
+			printLabels, _ := generateLabelString(toCheck.labelSelector)
```

Comment:
Since CheckReconciliation cannot return (true, err), this else is useless. Besides, if `err != nil`, I would log the error at least.